test/integration/consul-container/test/gateways/http_route_test.go

```diff
@@ -445,17 +503,43 @@ func TestHTTPRouteParentRefChange(t *testing.T) {
 	if testing.Short() {
 		t.Skip("too slow for testing.Short")
 	}
+
 	t.Parallel()
 
 	// infrastructure set up
 	address := "localhost"
 
-	listenerOnePort := 6000
-	listenerTwoPort := 6001
+	listenerOnePort := 6014
+	listenerTwoPort := 6015
+	serviceHTTPPort := 6016
+	serviceGRPCPort := 6017
```

Comment:
Could we use the magic behavior of `0` (preferably, but guessing maybe not possible) or perhaps https://github.com/hashicorp/consul/tree/main/sdk/freeport to automatically select a port for these, rather than manually avoiding overlap between tests?

Reply:
Unfortunately not, these aren't actually ports we're binding on the host, they're ports that are bound inside the context of a container and then hole-punched to. We might be able to work with the assumption that the ports are random enough when using `freeport` (also since they're in the ephemeral range) that they're unlikely to be bound inside the container context as well, but I'd prefer to test that out some other time.